File: Smartify_Mobil/app/src/main/java/com/example/smartify/api/models/Product.kt
package com.example.smartify.api.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue

@Parcelize
data class Product(
    @SerializedName("_id")
    val id: String,
    val name: String,
    val images: List<String> = listOf(),
    val price: Double,
    val oldPrice: Double? = null,
    val categoryNames: List<String> = listOf(),
    val categoryPath: List<String>? = null,
    val category: String,
    val mainCategory: String? = null,
    val isNew: Boolean = false,
    val isPopular: Boolean = false,
    val stock: Int? = 0,
    val inStock: Boolean = true, // stock > 0 ise true olarak hesaplanabilir
    val description: String = "",
    val specifications: Map<String, String> = emptyMap(),
    val features: @RawValue Any? = null, // API'den farklı formatlarda gelebilir
    val active: Boolean = true,
    val createdAt: String = "",
    val updatedAt: String = ""
) : Parcelable {
    // API'den dönen veri format uyumsuzluklarını gidermek için yardımcı extension fonksiyonlar
    
    // Features'ı güvenli bir şekilde liste olarak almak için
    fun getFeaturesList(): List<String> {
        return when (features) {
            is List<*> -> features.filterIsInstance<String>()
            is String -> listOf(features)
            else -> emptyList()
        }
    }
    
    // Stok durumunu hesaplama
    fun getStockStatus(): Boolean {
        return stock?.let { it > 0 } ?: inStock
    }
    
    // İlk görsel URL'sini alma
    fun getFirstImageUrl(): String {
        return images.firstOrNull() ?: ""
    }
} 

File: Smartify_Web/backend/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    description: {
        type: String,
        required: true
    },
    price: {
        type: Number,
        required: true,
        min: 0
    },
    category: {
        type: String,
        required: true
    },
    mainCategory: {
        type: String,
        required: true
    },
    categoryPath: {
        type: [String],
        required: true
    },
    categoryNames: {
        type: [String],
        required: true
    },
    images: {
        type: [String],
        default: []
    },
    stock: {
        type: Number,
        required: true,
        min: 0,
        default: 0
    },
    isNew: {
        type: Boolean,
        default: false
    },
    isPopular: {
        type: Boolean,
        default: false
    },
    features: {
        type: [String],
        default: []
    },
    specifications: {
        type: Object,
        default: {}
    },
    active: {
        type: Boolean,
        default: true
    }
}, {
    timestamps: true
});

// Arama için index oluştur
productSchema.index({ name: 'text', description: 'text' });

module.exports = mongoose.model('Product', productSchema); 